flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/functions/aggfunctions/DenseRankAggFunction.java

```diff
@@ -67,8 +65,8 @@ public DenseRankAggFunction(InternalType[] orderKeyTypes) {
 	@Override
 	public Expression[] accumulateExpressions() {
 		Expression[] accExpressions = new Expression[1 + operands().length];
-		// sequence = if (lastValues equalTo orderKeys) sequence else sequence + 1
-		accExpressions[0] = ifThenElse(orderKeyEqualsExpression(), sequence, plus(sequence, literal(1L)));
+		// sequence = if (lastValues equalTo orderKeys and sequence != 0) sequence else sequence + 1
+		accExpressions[0] = ifThenElse(and(orderKeyEqualsExpression(), not(equalTo(sequence, literal(0L)))), sequence, plus(sequence, literal(1L)));
```

Comment:
inside `orderKeyEqualsExpression()`, we already compare the last_value with null which is the initial value. Why this conflicts with the order by key's value? Do you mean the order by key is null?

Reply:
you scan see the unit test i commit. it will be zero rank